test/e2e/kuma_deploy_universal_test.go

```diff
@@ -159,4 +162,51 @@ destinations:
 				return "should retry", errors.Errorf("should retry")
 			})
 	})
+
+	const iterations = 50
+
+	It("Should distribute requests cross zones", func() {
+		err := EchoServerUniversal("universal1", echoServerToken)(remote_1)
+		Expect(err).ToNot(HaveOccurred())
+		err = EchoServerUniversal("universal2", echoServerToken)(remote_2)
+		Expect(err).ToNot(HaveOccurred())
+
+		responses := 0
+		for i := 0; i < iterations; i++ {
+			stdout, _, err := remote_1.ExecWithRetries("", "", "demo-client",
+				"curl", "-v", "-m", "3", "--fail", "localhost:4001")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stdout).To(ContainSubstring("HTTP/1.1 200 OK"))
+			Expect(stdout).To(ContainSubstring("universal"))
+
+			if strings.Contains(stdout, "universal1") {
+				responses++
+			}
+		}
+
+		Expect(responses > 10).To(BeTrue())
+		Expect(responses < iterations-10).To(BeTrue())
```

Comment:
// given service in zone 1 and zone 2
err := EchoServerUniversal("universal1", echoServerToken, WithLocality(true))(remote_1)
...

// when executing requests from the zone 1
responses := 0
...

// then requests are distributed to both zones
Expect(responses).To(Equal(iterations))